.github/workflows/build.yml

```diff
@@ -0,0 +1,31 @@
+# Original file found from https://github.com/FabricMC/fabric-example-mod/blob/1.17/.github/workflows/build.yml
+# Copyright to the respective authors. Modified to fit this project's needs.
+name: build
+on: [push, pull_request]
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [12, 13, 14, 15, 16]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+      runs-on: ${{ matrix.os }}
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v2
+        - name: Install Gradle
+          uses: gradle/wrapper-validation-action@v1
+        - name: Install JDK ${{ matrix.java }}
+          uses: actions/setup-java@v2
+          with:
+            java-version: ${{ matrix.java }}
+        - name: Build Gradle
+          if: ${{ runner.os != 'Windows' }}
+          run: chmod +x ./gradew
+        - name: Build Repository
+          run: ./gradlew build
+        - name: Build Artifacts
+          if: ${{ runner.os == 'Linux' && matrix.java == '16' }}
```

Comment:
Should this build artifacts phase be in this action? Or should we split out the build testing from the artifact uploading?

Reply:
I think perhaps later on we should split it out. For now, it's pretty simple to consolidate.

Reply:
Understood. LGTM